hack/update-openapigen.sh

```diff
@@ -18,36 +18,36 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KNOWN_VIOLATION_EXCEPTIONS=hack/violation_exceptions.list
-CURRENT_VIOLATION_EXCEPTIONS=hack/current_violation_exceptions.list
-OPENAPI_SPEC_FILE=pkg/openapi/openapi_generated.go
+# find_project_root [start_dir] [marker]
+#   start_dir : directory to begin the search (defaults to this script’s dir)
+#   marker    : filename or directory name to look for (defaults to "go.mod")
+#
+# Prints the first dir containing the marker, or exits 1 if none found.
+find_project_root() {
```

Comment:
I am aware we have it in `common.sh` for our openshift-ci setup scripts, but I want to push it upstream too so decided to make the script self-contained.